client/ayon_core/pipeline/editorial.py

```diff
@@ -281,3 +281,32 @@ def get_media_range_with_retimes(otio_clip, handle_start, handle_end):
         returning_dict.update(version_data)
 
     return returning_dict
+
+
+def create_clip(path, name, frames, framerate):
+    range = otio.opentime.TimeRange(
+        start_time=otio.opentime.RationalTime(0, framerate),
+        duration=otio.opentime.RationalTime(frames, framerate)
+    )
+
+    media_reference = otio.schema.ExternalReference(
+        available_range=range,
+        target_url=f"file://{path}"
+    )
+
+    return otio.schema.Clip(
+        name=name,
+        media_reference=media_reference,
+        source_range=range
+    )
+
+
+def export_otio(clips_data, output_path):
+    clips = []
+    for name, data in clips_data.items():
+        clips.append(
+            create_clip(data["path"], name, data["frames"], data["framerate"])
+        )
+
+    timeline = otio.schema.timeline_from_clips(clips)
+    otio.adapters.write_to_file(timeline, output_path)
```

Comment:
Isn't the entry point here much more sensible to be actual `otio.schema.Clip` entries?

This would make the code that generates the data much more readable.
I'm also a bit unsure why the clips data needs to be a dict by name. It's quite limiting what `export_clips` would support since it cannot ever export the same clip name then.

I think this is totally fine code in the Export OTIO right?

```python
clips = []
for representation in representations:
    version = self._version_by_representation_id[representation["_id"]]
    name = self._version_path_by_id[version["_id"]]

    clip = create_clip(
        path=get_representation_path_with_anatomy(
            representation, anatomy
        ),
        name=
        frames=(
            version["data"]["frameEnd"] -
            version["data"]["frameStart"]
        ),
        framerate=version["data"]["fps"]
    )
    clips.append(clip)

export_clips(clips, output_path)
```

Also, be aware that by adding this global `export_otio` loader that you might be getting import errors on hosts that e.g. do not have OTIO available since it imports `from ayon_core.pipeline.editorial import export_otio`.

Reply:
Yeah, maybe I'll push the lib code to the action. Think eventually we'll need to push some code out of it, but that can come in later PRs.